### DISCUSSION TITLE
Error when trying to run fresh-start.sh

### DISCUSSION THREAD

**Discussion by Menelion** [evertramos/nginx-proxy-automation#365] [Q&A]
Hello everyone,

I'm desperate. Was trying to hook everything up on my own, then thought this solution would solve my problems. However, no dice. When I try to run _fresh-start.sh_ on WSL with SUDO, it asks me for the service names and then shows this:

```shell
 ✘ ERROR   It seems there is no default image or version, please check the .env file at '/mnt/d/Repos/nginx-proxy-automation/bin'
```

I did recheck both _.env_ files, everything is OK. The most frustrating thing it doesn't say which image it lacks.

Thanks for your help.

**Comment by Menelion** [comment]
After some debugging I found out that it complains about the LetsEcnrypt image version. I tried everything: `latest`, `"2.1"`, `"2.2"`, tried editing local _.env_ and _/bin/.env_, still no dice. It complains about LetsEncrypt default image name or version.

My local _.env_:

```shell
PID_FILE=.fresh_start.pid
ALLOW_RUN_WITH_SUDO=true
NGINX_WEB_SEVICE_NAME=proxy-web
NGINX_IMAGE_VERSION=latest
DOCKER_GEN_SEVICE_NAME=docker-gen
DOCKER_GEN_IMAGE_VERSION=latest
LETS_ENCRYPT_SEVICE_NAME=acme-companion
NGINX_PROXY_COMPANION_IMAGE_VERSION=latest
IPv4=XXX.XXX.XXX.XXX
IPv6=::0
NETWORK=proxy
NGINX_FILES_PATH=/srv/docker/nginx
NGINX_WEB_LOG_DRIVER=json-file
NGINX_WEB_LOG_MAX_SIZE=2m
NGINX_WEB_LOG_MAX_FILE=10
NGINX_GEN_LOG_DRIVER=json-file
NGINX_GEN_LOG_MAX_SIZE=2m
NGINX_GEN_LOG_MAX_FILE=10
NGINX_LETSENCRYPT_LOG_DRIVER=json-file
NGINX_LETSENCRYPT_LOG_MAX_SIZE=2m
NGINX_LETSENCRYPT_LOG_MAX_FILE=10
DOCKER_HTTP_=80
DOCKER_HTTPS=443
#SSL_POLICY=Mozilla-Modern
DEFAULT_EMAIL=valid.email@domain.tld
DEFAULT_HOST=my.domain.tld
DOCKER_HOST_ROOTLESS_PATH=
```

**Reply by evertramos** [reply]
Did you run the script with '--skip-docker-image-check'?

´´´bash
cd proxy/bin && ./fresh-start.sh --yes -e your_email@domain --skip-docker-image-check
```

**Comment by Menelion** [comment]
Thank you. Now it's running but the certificate is invalid.

![image](https://user-images.githubusercontent.com/595597/209716967-b7dad615-3ec8-4b4b-afc6-e576d883e6c9.png)

Also a stupid question (I did read the instructions, honestly!), how do I run multiple services on multiple subdomains with the proxy? I would like to run GitLab and Redmine on different subdomains. Thank you!

**Reply by evertramos** [reply]
You can run multiple sites just like that:

```bash
docker run -d -e VIRTUAL_HOST=your.domain.com \
            -e LETSENCRYPT_HOST=your.domain.com \
            -e LETSENCRYPT_EMAIL=your.email@your.domain.com \
            --network=proxy \
            --name my_app \
            httpd:alpine
```

Run any service, remove the binding port and add the environment variables above to your service.

About the certificates you must check the letsencrypt logs to see what is wrong with it.

Also check:

https://github.com/steevepay/gitlab-docker-letsencrypt
